### PR TITLE
Use ExoApi v2

### PR DESCRIPTION
**Pull request by shoekstra**
This drops support for the deprecated v1 version of the ExoAPI; if you still need 
to use that you'll need to stick with v0.1.0 of this provider.

### CODE REVIEW

**Comment by borisroman** [comment]
@shoekstra It looks good to me, except for;
- The acceptance test fails `The email address *** already exists.`
- Should a piece of documentation for the required user actions be added for this PR? In order for users to have to upgrade go smoothly?

File: internal/mcaf/o365_client.go [hunk 1]
```diff
@@ -4,13 +4,16 @@ import (
...
+	"io"
 	"io/ioutil"
 	"log"
 	"net/http"
 	"net/http/httputil"
 	"net/url"
 	"sync"
 	"time"
+
+	"github.com/google/go-querystring/query"
```
Comment (shoekstra) [inline internal/mcaf/o365_client.go:16]: This is great, didn't know about this...

**Comment by shoekstra** [comment]
Hey @borisroman, 

I reran the test after removing the existing alias and it now passes; still have to write a [sweeper](https://www.terraform.io/docs/extend/testing/acceptance-tests/sweepers.html) so that if a previous test fails the alias will get removed before running fresh tests.

Re upgrade instructions, this provider maintains the same user API so there is no change to the [setup instructions](https://registry.terraform.io/providers/schubergphilis/mcaf/latest/docs). It should be a case of using 0.2 or newer of this provider and specifying the v2 URL and credentials.

**Comment by shoekstra** [comment]
Reran the tests and still good so going to go ahead and merge this.

**Pull request merged**